tests/test_e2e.py

```diff
@@ -124,20 +124,21 @@ def test_username_password(self):
         self.skipUnlessWithConfig(["client_id", "username", "password", "scope"])
         self._test_username_password(**self.config)
 
-    def _get_app_and_auth_code(self):
+    def _get_app_and_auth_code(self, **kwargs):
         return _get_app_and_auth_code(
             self.config["client_id"],
             client_secret=self.config.get("client_secret"),
             authority=self.config.get("authority"),
             port=self.config.get("listen_port", 44331),
             scopes=self.config["scope"],
-            )
+            **kwargs)
 
     def test_auth_code(self):
         self.skipUnlessWithConfig(["client_id", "scope"])
-        (self.app, ac, redirect_uri) = self._get_app_and_auth_code()
+        nonce = "foo"
```

Comment:
Since nonce is an optional parameter, it makes sense to have a new test case for api calls with nonce parameter. So that our no-nonce test case is still there and we have a new one with nonce :)

Reply:
The test case you see here, is a test case for AAD. We still have other similar test cases for B2C and ADFS. Those test cases still operate in a no-nonce way (they may or may not support nonce anyway).

Besides, if our goal is to see whether we (i.e. MSAL) implement that *optional* logic correctly, then this PR itself serves as a white-box test already. We have 3 people reviewing this 19-line PR (and most of them are docs and tests), this is already a record-setting optimal reviewer-per-line ratio. I have high confidence on your reviews. :-)

Reply:
I know there are no-nonce test cases for B2C and ADFS but they serve the purpose to test B2C and ADFS. I just think instead of combining this test with our already existing test case, it will be better for debugging later if there is an issue relating this area.

Reply:
Ended up adding a test case for mismatching nonce, and retain the without-nonce test case. So we have 3 test cases now.